mods/eradicate-immersive-menus.wh.cpp

```diff
@@ -0,0 +1,353 @@
+    settings.network        = Wh_GetIntSetting(L"network",        TRUE);
+    settings.defender       = Wh_GetIntSetting(L"defender",       TRUE);
+    settings.nosettingsicon = Wh_GetIntSetting(L"nosettingsicon", TRUE);
+#endif
+}
+
+BOOL Wh_ModInit()
+{
+    LoadSettings();
+
+    /* Explorer is always 64-bit */
+#ifdef _WIN64
+    WCHAR szProcessPath[MAX_PATH];
+    GetModuleFileNameW(NULL, szProcessPath, MAX_PATH);
+    BOOL bIsExplorer = (wcsicmp(szProcessPath, L"C:\\Windows\\explorer.exe") == 0);
+
+    if (settings.nosettingsicon)
+    {
+        Wh_SetFunctionHook(
+            (void *)SetMenuItemInfoW,
+            (void *)SetMenuItemInfoW_hook,
+            (void **)&SetMenuItemInfoW_orig
+        );
+    }
+#endif
+
+    if (settings.explorer
+#ifdef _WIN64
+    && wcsicmp(szProcessPath, L"C:\\Windows\\System32\\SecurityHealthSystray.exe")
+#endif
+    )
+    {
+        if (!HookICMH_CAODTM(
+            L"shell32.dll",
+            &ICMH_CAODTM_orig_shell32
+        ))
+        {
+            return FALSE;
+        }
+        
+        
+        if (!HookICMH_CAODTM(
+            L"ExplorerFrame.dll",
+            &ICMH_CAODTM_orig_ExplorerFrame
+        ))
+        {
+            return FALSE;
+        }
+        
+        /* Explorer itself is always 64-bit */
+#ifdef _WIN64
+        if (bIsExplorer)
+        {
+            if (!HookICMH_CAODTM(
+                NULL,
+                &ICMH_CAODTM_orig_explorer
+            ))
+            {
+                return FALSE;
+            }
+
+            if (!HookICMH_CAODTM(
+                L"twinui.dll",
+                &ICMH_CAODTM_orig_twinui
+            ))
+            {
+                return FALSE;
+            }
+
+            if (!HookICMH_CAODTM(
+                L"twinui.pcshell.dll",
+                &ICMH_CAODTM_orig_twinui_pcshell
+            ))
+            {
+                return FALSE;
+            }
+        }
+#endif
+    }
+
+    /* Tray icons are always 64-bit */
+#ifdef _WIN64
+    if (bIsExplorer)
+    {
+        if (settings.sound)
+        {
+            if (!HookICMH_CAODTM(
+                L"SndVolSSO.dll",
+                &ICMH_CAODTM_orig_SndVolSSO
+            ))
+            {
+                return FALSE;
+            }
+        }
+
+        if (settings.network)
+        {
+            if (!HookICMH_CAODTM(
+                L"pnidui.dll",
+                &ICMH_CAODTM_orig_pnidui
+            ))
+            {
+                return FALSE;
+            }
+        }
+    }
+
+    if (settings.defender
+    &&  0 == wcscmp(wcslwr(szProcessPath), L"c:\\windows\\system32\\securityhealthsystray.exe"))
+    {
+        if (!HookICMH_CAODTM(
+            L"SecurityHealthSSO.dll",
+            &ICMH_CAODTM_orig_SecurityHealthSSO
+        ))
+        {
+            return FALSE;
+        }
+    }
+#endif
+
+    return TRUE;
+}
+
+BOOL Wh_ModSettingsChanged(BOOL *bReload)
+{
+    *bReload = TRUE;
+    LoadSettings();
```

Comment:
Doesn't matter all that much, but just FYI - this isn't necessary since the mod will be reloaded right away.